levelkdev/swapr-dapp
# Feature/bridge swap

# Summary

Closes #1187 

`Bridge Swap` is done by `Socket API`. Previous implementation of `Socket` didn't change except small improvements in `getBridgingMetadata`.

`Bridge Swap` mostly works by `isBridgeSwapActive` argument which we keep inside `ecoBridge` store. This argument helps handling `validation` and selectors e.g `supportedBridges`, `lists`.

I had to add new props to currency components like:
- `isOutputPanel` - this is necessary to separate `fromTokens` and `toTokens` (it's optional props so it doesn't affect swap or normal bridge)
```js
const tokens = isOutputPanel ? allTokensOnSecondChain : allTokens
```
- `hideBalance` - `CurrencyList` component supports getting balance of token only on `activeChainId` so on bridge on `toChainId` we got infinite loading of token balance


 # To Test

1. Open the page `Bridge`
- [ ] - Check if `Bridge Swap` tab is visible and works correctly
- [ ] - Check if `output panel` shows up when bridge swap is active
- [ ] - `Select token` button should be disable and has spinner when `from` or `to` chain changed
- [ ] - Tokens should be updated after chain changed
- [ ] - Do a `bridge swap` transaction (NOTE: sometimes `Socket` cannot find a route (unsupported pair or too small amount of token))
- [ ] - Check `status` of this transaction
    - firstly status should be `pending`
    - after a few minutes it should change to `confirmed`
- [ ] - After transaction check your balance on second chain
